[UI/UX] Use Tabs on Game Settings

flavioislima opened the pull request:
This replaces the Collapsable elements with a Tabbing approach on the Game Settings.
It also remembers the last used tab for each game so makes our life easier when testing different configs.

![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/26871415/98eaef10-e5e3-4656-ba28-d9538b6f5e67)

![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/26871415/e6c5cae1-0c3f-4820-bc4c-1678b59c358b)

![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/26871415/f4a1d182-9449-4dfe-9e2f-a3ca93e863d2)

![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/26871415/147b0478-e832-458a-89c4-2e1ded0d6fc7)


---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

arielj commented on src/frontend/screens/Settings/components/SettingsModal/index.scss:
```diff
@@ -2,8 +2,8 @@
   flex-direction: column;
-  justify-content: space-between;
   max-width: 800px;
+  min-height: 50vh;
```
maybe we can also position the modal a fixed amount from the top? now changing tabs changes the height and it keeps jumping around, it's not great for accessibility that much movement

flavioislima replied:
good idea, that was bothering me as well. I will try.

flavioislima replied:
done!

arielj commented on src/frontend/screens/Settings/sections/GamesSettings/index.tsx:
```diff
@@ -82,101 +118,75 @@ export default function GamesSettings({ useDetails = true }: Props) {
+            <ShowFPS />
+            <Mangohud />
+            <BattlEyeRuntime />
+            <EacRuntime />
```
this is showing on Mac but I understand EAC doesn't work on Mac (I haven't checked yet if it's an issue also on main though)

Etaash-mathamsetty replied:
both runtimes are linux exclusive

flavioislima replied:
fixed

arielj commented:
Looks like the change to a different place broke the alignment for the text of these buttons

<img width="430" alt="Screenshot 2023-10-01 at 4 22 59 PM" src="https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/188464/bd6dfb4c-2112-4a05-9552-97096af97792">

arielj commented:
Going to the General Settings > Game Defaults crashes with:
<img width="558" alt="Screenshot 2023-10-01 at 4 25 14 PM" src="https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/188464/dd6f9ec6-a109-4df2-819e-0f2f74b545b9">

I guess it's for the tab remembering thingy?

flavioislima commented:
> Looks like the change to a different place broke the alignment for the text of these buttons
> <img alt="Screenshot 2023-10-01 at 4 22 59 PM" width="430" src="https://user-images.githubusercontent.com/188464/271849292-bd6dfb4c-2112-4a05-9552-97096af97792.png">

Fixed and Also added a small fading animation to when they are running:
![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/26871415/b3d39e23-075b-4ca2-a0a5-f0881790aea6)

flavioislima commented:
> Going to the General Settings > Game Defaults crashes with: <img alt="Screenshot 2023-10-01 at 4 25 14 PM" width="558" src="https://user-images.githubusercontent.com/188464/271849370-dd6f9ec6-a109-4df2-819e-0f2f74b545b9.png">
> 
> I guess it's for the tab remembering thingy?

Fixed that as well.

imLinguin commented:
![obraz](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/62100117/7ab54b06-fcd9-4793-9d4c-6ab5b94387c0)

Let's make Wine optional tab for native games :D

flavioislima commented:
> ![obraz](https://user-images.githubusercontent.com/62100117/272003131-7ab54b06-fcd9-4793-9d4c-6ab5b94387c0.png)
> 
> Let's make Wine tab hidden for native games :D
> 
> Same goes for some options in different tabs, like anti cheat runtimes
> 
> ![obraz](https://user-images.githubusercontent.com/62100117/272003429-aa156b26-e0e8-4dfd-887e-3a244d7cc2d0.png)

Fixed.
Also changed the index to strings because using numbers was causing other issues when the Wine tab was not available.

arielj commented on src/frontend/screens/Settings/components/Tools/index.tsx:
```diff
@@ -98,6 +105,12 @@ export default function Tools() {
     ev.preventDefault()
   }
 
+  console.log('toolsSettings', {
+    winetricksRunning,
+    winecfgRunning,
+    runExeRunning
+  })
```
```suggestion
```

arielj replied:
to delete this

arielj commented on src/frontend/screens/Settings/components/Tools/index.tsx:
```diff
@@ -131,12 +144,15 @@ export default function Tools() {
+            <div>
+              {t('setting.runexe.title')} <br />
+              <span>{t('setting.runexe.message')}</span>
+            </div>
```
we don't need this div, we can also remove the new css, we need to add a `justify-content: center` to `.drag` fix the alignment of the run exe on prefix button

flavioislima replied:
fixed as well

arielj commented on src/frontend/screens/Settings/components/Tools/index.scss:
```diff
@@ -20,7 +27,12 @@
     white-space: nowrap;
 
-    & > span {
+    & div {
+      position: relative;
+      top: -18px;
+    }
```
```suggestion
    justify-content: center;
```

(I left another comment to remove the div, and with this css it fixes the alingment)

flavioislima replied:
fixed

arielj commented on src/frontend/screens/Settings/sections/GamesSettings/index.tsx:
```diff
@@ -82,101 +116,81 @@ export default function GamesSettings({ useDetails = true }: Props) {
+        )}
+      </Tabs>
+
+      <TabPanel value={value} index={'wine'}>
```
we can remove the fragment after this line now

flavioislima replied:
fixed

arielj reviewed (commented):
I left a few comments, but it works good for me and looks way better.

I'll wait for the change setting a fixed top distance and it's all good from me

flavioislima commented:
Addressed all comments, I think looks good now.

arielj reviewed (approved):
Looks good to me :+1:

arielj reviewed (changes requested):
wait, I found a bug opening settings for a browser app sideloaded 

![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/188464/bba7d65f-bcc3-4dc6-ad07-15e4d7ca77b1)

arielj commented:
(well I guess we have a bigger problem with those apps, we are showing a bunch of settings that makes no sense, maybe the settings button for sideloaded apps should open the edit app modal instead of settings, no setting applies for theme)

arielj commented:
That same error happens for any native linux game since the `wine` tab doesn't exist. I imagine it's the same for native Mac games too

arielj commented:
when that happens, there's no `Wine` tab, but the wine tab content is displayed 
![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/188464/3cd70310-7066-4bba-96bf-b569fb7e8d6b)

flavioislima commented:
@arielj ooff, the burden of supporting so many configurations I guess. lol
will check, thanks for testing!

flavioislima commented:
@arielj removed the settings button for browser games for now until we have proper settings for them (maybe user-agent, open fullscreen by default, etc) in the future.
Improved the logic to get more use cases and tested on macOS, Windows and Linux, native, sideloaded, wine, etc.
Also the Default game settings on each platform
Could not find any other issue so far.

arielj reviewed (approved):
Looks good to me :+1: 

I tested only on linux though, but I haven't seen any other issue.
